### ISSUE TITLE
It should be possible to get rid of creatures

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1143]
Creatures will enter your dungeon at random, and because you can only have a limited amount, they have different uses and a specific composition is needed to get more powerful creatures you should be able to get rid of creatures you don't want.

My suggestion is to do so like in the dungeon keeper games, and be able to drop creatures into the portal to make them leave.

**Comment by hwoarangmy** [comment]
Yes, that has been discussed some time ago and I guess it would be a nice feature. Until now, most of us prefer to slap them to death instead of letting them go ^^

Note that implementing that is pretty straight forward

**Comment by eugeneloza** [comment]
Moreover, if you have a crypt, slapped creature's bodies can rot and turn into something useful :)

**Comment by Loobinex** [comment]
I'd say that is abuse and should be discouraged, not encouraged.
In DK slaps were a micro-management mechanic, increasing the speed/effectiveness of slapped creatures in combat/research/everything by 25%, on top of it being fun to do. You'd needed many, many slaps to kill a creature. As a result slapping creatures was a good thing to do.

What I see so far in OpenDungeons is that it takes just a few slaps to kill, and a single slap hurts a lot making it not a wise move to slap the creatures you want to keep, which is a shame.

**Comment by Bertram25** [comment]
> What I see so far in OpenDungeons is that it takes just a few slaps to kill, and a single slap hurts a lot making it not a wise move to slap the creatures you want to keep, which is a shame.

I think @Loobinex has a point. So I guess we have an issue about setting the number of slaps to kill a creatures much higher. 

I suppose we then have the issue to be able to get rid of the creatures. In my humble, we could use two ways:
- Put the creature in a portal, so that it gets away with a protection that if the player drops many creatures in the portal within a turn the creatures do not get discarded as a careless player could remove his whole army doing this.
- Add a skill permitting to drop a creature on the temple (dungeon heart) to earn random effects, to be defined. E.g: make every creatures move faster for some turn, earn XP, gold, mana, ... depending on the creature.

@OpenDungeons/developers WDYT?

**Comment by Loobinex** [comment]
As far as I know there's nothing to protect the player from discarding his entire army through the portal in either dungeon keeper game, and I have never seen any complains about it.

**Comment by Bertram25** [comment]
> As far as I know there's nothing to protect the player from discarding his entire army through the portal in either dungeon keeper game, and I have never seen any complains about it.

We could try without the "protection" first, indeed.

**Comment by hwoarangmy** [comment]
I would say the only protection we might need is to check if there are enemies in sight nearby. If not, the keeper wants to discard the unit. If yes, it should fight instead.

**Comment by Loobinex** [comment]
That should be consistent with how all room assignments are done. If a creature is dropped into a room it is assigned to work that room, unless an enemy is nearby. Being assigned to the portal means leaving the dungeon, the same way being assigned to the training rooms means 'go train'.

**Comment by Bertram25** [comment]
> I would say the only protection we might need is to check if there are enemies in sight nearby. If not, the keeper wants to discard the unit. If yes, it should fight instead.

Good idea but we'd need to define "nearby" then. WDYT about the visible enemies within 15 tiles?

**Comment by Loobinex** [comment]
In the dungeon keeper games, the standard combat range is checked. So on drop first check for combat, only afterwards go into room assignment.

**Comment by hwoarangmy** [comment]
> Good idea but we'd need to define "nearby" then. WDYT about the visible enemies within 15 tiles?

Well, ATM, when a creature is dropped in a room, the room can handle the creature and push fitting actions. If the creature sees enemies, it will stop what it is doing and fight. That means that the protection might already fit the current architecture and there might be no need to do anything. And in this case, the vision of the creature would be used (15 for most creatures). Note that a problem might occur if we implement a spell that reduces creatures vision.

**Comment by Bertram25** [comment]
> Note that a problem might occur if we implement a spell that reduces creatures vision.

Well, let's see when the case happens. :) I'm fine with the rest.

**Comment by eugeneloza** [comment]
Kinda https://youtu.be/8gInoX2mwgM?t=8m33s (however, sacrifice altar will require additional room)
